src/components/onboarding/steps/collector_intent/index.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react"
+import styled from "styled-components"
+
+import Step from "../step"
+import Option from "./option"
+
+const OptionsContainer = styled.div`
+  width: 450px;
+  margin: 0 auto 100px;
+`
+
+interface Props {
+  onStateChange?: ({ nextButtonEnabled }) => void
+}
+
+interface State {
+  selectedOptions: { [option: string]: boolean }
+  selectedCount: number
+}
+
+class CollectorIntent extends React.Component<Props, State> {
+  options = [
+    "Buy Art & Design",
+    "Sell Art & Design",
+    "Research Art Prices",
+    "Learn About Art",
+    "Find Out About New Exhibitions",
+    "Read Art Market News",
+  ]
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      selectedOptions: {},
+      selectedCount: 0,
+    }
+  }
+
+  onOptionSelected = (index, selected) => {
+    const option = this.options[index]
+
+    let selectedOptions = this.state.selectedOptions
+    selectedOptions[option] = selected
+
+    let count = 0
+    for (let key in selectedOptions) {
+      if (selectedOptions[key]) count++
+    }
+
+    this.setState({ selectedOptions, selectedCount: count })
+    this.props.onStateChange({ nextButtonEnabled: count > 0 })
```

Comment:
Would be worth exploring Redux here, because I bet this will only scale outwards with time. Then there's no need to pass things from component to component because you can just 

```js
@connect(state => ({
  selectedOptions: state.onboarding.selectedOptions
})
class MyComponent extends Component {
   ... 
} 
```
and keep things loosely coupled. There is a lot of FUD around Redux's boilerplate, but @sweir27 built something similar to this recently and it worked really well. Might be worth pinging her for more information on the overall flow.

Reply:
I agree that it’s likely that this state machine will grow in the future and this is one of the few situations where I think the amount of dealing with state would benefit from using redux, but I’m not sure we should add it right away, it’s probably straight forward for now. So I’d say we convert to redux either once it gets very complicated very soon or once we’ve delivered the initial version for the test and we start work on the fully featured and flexible on-boarding framework.

Reply:
FWIW I wouldn't wait tooooo long. I've found that it's not too bad to get set up and once you _are_ set up, it increases productivity/testability/etc. immensely!

Reply:
This is totally off the top of my head, but once you have the basic skeleton (which isn't much) your app will pretty much scale out infinitely: 

```javascript
// actions.js 

const actions = {
  NEXT_PAGE: 'NEXT_PAGE',
  PREVIOUS_PAGE: 'PREVIOUS_PAGE'
}

const nextPage = () => ({
  type: actions.NEXT_PAGE
})

const previousPage = () => ({
  type: actions.PREVIOUS_PAGE
})


// reducer.js 

const initialState = {
  page: 0,
  totalPages: 10
}

const onboardingReducer = (state = initialState, action) => {
  switch (action.type) {
    case actions.NEXT_PAGE: {
      const page = state.page + 1
      const nextButtonEnabled = page < state.totalPages 

      return u({
        page,
        nextButtonEnabled
      }, state)
    }

    case actions.PREVIOUS_PAGE: {
      const page = state.page - 1
      const nextButtonEnabled = page > 0

      return u({
        page,
        nextButtonEnabled
      }, state)
    }
}

const reducers = composeReducers({ 
  onboarding: onboardingReducer
})

const middleware = [
  createLogger({ 
    collapsed: true 
  })
]

const store = createStore(reducer, middleware) 

// App.js

const App = () => 
  <Provider store={store}>
    <Layout />
  </Provider>

// Layout.js

@connect(state => ({
  nextButtonEnabled: state.onboarding.nextButtonEnabled,
  page: state.onboarding.page
}), {
  nextPageAction: nextPage,
  previousPageAction: prevPage
})
class Layout extends Component {

  render () {
    const {
      nextButtonEnabled, 
      nextPageAction,
      previousPageAction,
      page
    } = this.props

    return (
      <div>
        <div>
          Hello, you are on step {page}
        </div>

        <NextButton 
          enabled={nextButtonEnabled} 
          onClick={nextPageAction}
        />

        <PreviousButton 
          onClick={previousPageAction}
        />
      </div>
    )
  }
}

// Further down in the tree, other components can just "pull" from store state via @connect
// which makes things very portable and light-weight, lending to more refined (and rapid)
// refactors later on. 

```

Like @sweir27 mentioned above, a little goes a very long way and untangling later (since you know it _will_ be scaling in complexity and scope) might only serve to discourage refactors. 

Anyways, just my 2 cents! Carry on 😄

Reply:
I didn't know about the `@connect` annotation, that's a game changer...

Reply:
Yeah! pretty much any higher-order-component can be used as a decorator. For example, [this bit here](https://github.com/artsy/force/blob/master/desktop/apps/consign/components/sign_up/index.js#L110) can easily be rewritten like 

```javascript
@reduxForm({
  form: 'signUp', 
  validate
}),
@connect(
  mapStateToProps,
  mapDispatchToProps
)
export default class SignUp extends Component {
  ... 
}
```

While I'm here (sorry sorry...) another thing I want to emphasize is now nice Redux's [Logger middleware](https://github.com/evgenyrodionov/redux-logger) is to use during development. There's literally zero ambiguity about whats going on when and where and this immediately translates into increased development speed.